include/grpcpp/security/server_credentials.h

```diff
@@ -89,16 +87,7 @@ class ServerCredentials : private grpc::internal::GrpcLibrary {
   // TODO(dgq): the "port" part seems to be a misnomer.
   virtual int AddPortToServer(const std::string& addr, grpc_server* server) = 0;
```

Comment:
Similar to my comments elsewhere about removing the wrapping subclasses on the client side, I think this base class should have a data member for the C-core creds and should have default implementations for these methods, so that subclasses need to override the methods only if they want non-default behavior, which should be a subset of cases (and most of that subset should go away once we implement transport selection in C-core).

Reply:
Done. The base class is now abstract with some state mirroring the old insecure credentials, and subclasses implement a subset of functionality.